libs/wingc/src/type_check.rs

```diff
@@ -2122,6 +2118,7 @@ impl<'a> TypeChecker<'a> {
 					};
 				}
 
+				if let Some(constructor) = constructor {
```

Comment:
something with the indentation here is wrong, it also seems like there's code duplication here (the comment " Type check constructor" appears twice). Can you take a look and either explain or fix this so it'll be easier to review..